platform/mbed_power_mgmt.h

```diff
@@ -205,6 +205,32 @@ static inline void system_reset(void)
 {
     NVIC_SystemReset();
 }
+ 
+/** Provides the time spent in sleep mode, since system is up and running
+ *
+ *  @return  Time spent in sleep
+ */
+uint64_t mbed_time_sleep(void);
+
+/** Provides the time spent in deep sleep mode, since system is up and running
+ *
+ *  @return  Time spent in deep sleep
+ */
+uint64_t mbed_time_deepsleep(void);
+
+/** Provides the time spent in idle thread since the system is up
+ *
+ * @return  Idle thread time.
+ */
+uint64_t mbed_time_idle(void);
+
+
+/** Provides the time since the system is up and running
+ *
+ * @return  System uptime.
+ */
+uint64_t mbed_uptime(void);
```

Comment:
We have `clock()` in retarget that is similar to this, should that be updated to use this new functionality (clock is implemented with us ticker read that is not ideal).

Reply:
@0xc0170 - Good point. I believe updating will be good, when we have lp ticker for all targets.
Else we can use lp ticker as default and us ticker as fall back option. If we agree on this, then i will create new PR for this change

Reply:
Sounds good to me

Reply:
`clock()` is supposed to be "CPU run time", not "uptime". What it actually is on mbed OS I think depends on platform config. I believe because it uses us_ticker and doesn't lock it, it's sort of "runtime" if the platform ends up stopping us_ticker during idle. If you can get a real "non-idle" time from the RTOS, that's what clock should really be returning.

The "uptime" is accessible now as `Kernel::get_ms_count()` - that's defined as counting up from 0, and will include any "sleep because of idle" time (because ticks are corrected after idle sleeps).

Reply:
@kjbracey-arm - Thanks for clarifying, that clock is "run time", and it makes sense to leave it as us_ticker.